baseplate/lib/metrics.py

```diff
@@ -274,9 +274,11 @@ def __init__(self, transport: Transport, name: bytes):
 
         self.start_time: Optional[float] = None
         self.stopped: bool = False
+        self.sample_rate = 1.0
 
-    def start(self) -> None:
+    def start(self, sample_rate: float = 1.0) -> None:
         """Record the current time as the start of the timer."""
+        self.sample_rate = sample_rate
```

Comment:
nit: can this go down with `self.start_time = ` so it's below the guard statements as well?

Reply:
👍